File: Regression/Scenarios/Scripts/HIV/plotBinnedPopulation.R
# SUMMARY: Plot the population by age group
# INPUT ARGUMENTS: (1) Path to BinnedReport.json
# OUTPUT: figs/PopulationByAge.png

rm( list=ls( all=TRUE ) )
graphics.off()

library(jsonlite)
library(reshape)
library(ggplot2)

fig_dir = 'figs'
if( !file.exists(fig_dir) ) {
    dir.create(fig_dir)
}

args = commandArgs()

idx = which( args == '--args' )
if( length(idx) == 0 || length(args) < idx + 1) {
    stop('USAGE: Rscript plotBinnedPopulation.R BinnedReport.json')
}
fn = args[idx+1]
J = fromJSON(fn)

mat = t(J$Channels$Population$Data)
colnames(mat) = J$Header$Subchannel_Metadata$MeaningPerAxis
m = melt(mat)

p = ggplot( m, aes(x=X1, y=value, colour=factor(X2), group=factor(X2)) ) + 
    geom_line() +
    scale_colour_discrete(breaks=J$Header$Subchannel_Metadata$MeaningPerAxis, name="Age Group") +
    xlab( "Simulation Day" ) +
    ylab( "Population" ) +
    ggtitle( "Population by Age" )

outfile = file.path(fig_dir,"PopulationByAge.png")
print(paste('Saving figure to', outfile))
png( outfile, width=640, height=480)
print(p)
dev.off()

File: Regression/Scenarios/Scripts/HIV/plotBinnedPopulation.py
#!/usr/bin/python

# SUMMARY: Plot the population by age group
# INPUT ARGUMENTS: (1) Path to BinnedReport.json
# OUTPUT: figure drawn on screen


from __future__ import print_function
import argparse
import json
import matplotlib.pyplot as pyplot


def main(filename, channel):

    with open(filename, 'rb') as handle:
        report = json.load(handle)

    labels = report['Header']['Subchannel_Metadata']['MeaningPerAxis'][0][0]
    data = report['Channels'][channel]['Data']

    figure = pyplot.figure('{0} by Age'.format(channel))
    pyplot.axes([0.1,0.1,0.71,0.8])
    for (trace, label) in zip(data, labels):
        pyplot.plot(trace, label=label)

    pyplot.title('{0} by Age'.format(channel))
    pyplot.xlabel('Simulation Day')
    pyplot.ylabel('Population')
    pyplot.legend(loc=(1.03, 0.135), prop={"size":10})

    pyplot.show()

    pass


if __name__ == '__main__':
    parser = argparse.ArgumentParser()
    parser.add_argument('filename', default='BinnedReport.json')
    parser.add_argument('-c', '--channel', default='Population')
    args = parser.parse_args()
    main(args.filename, args.channel)

    pass
